src/types.js

```diff
@@ -1,3 +1,24 @@
+function parseTimestamp(timestamp) {
+  // Postgres format 2001-01-01 00:00:00 or 2001-01-01 00:00:00.123456
+  // Postgres format +292278994-08-16 23:00:00 or -292275055-05-16 23:00:00 for Infinity values
+  // eslint-disable-next-line no-case-declarations
+  const [, year, month, day, hour, minute, second, millisecond = ''] = timestamp.match(
+    /^([+|-]?\d+)-(\d{1,2})-(\d{1,2}) (\d{1,2}):(\d{1,2}):(\d{1,2})(?:.(\d{1,3}))?(?:\d{1,3})?$/,
+  );
+
+  // Data Api may return timestamps out of range of JS max timestamp value of +-8640000000000000
+  // Year comparison is performed first and if the year is in range, then ISO date strings may be compared
+  const yearNumber = Number(year);
+  if (yearNumber > 275760) {
+    return Infinity;
+  }
+  if (yearNumber < -271821) {
+    return -Infinity;
+  }
```

Comment:
```suggestion
  const yearNumber = Number(year);
  const isoDateString = `${year}-${month}-${day}T${hour}:${minute}:${second}.${millisecond.padEnd(3, 0)}Z`;
  if (yearNumber > 275760 || isoDateString >= '+275760-09-13T00:00:00.000Z') {
    return Infinity;
  }
  if (yearNumber < -271821 || isoDateString <= '-271821-04-20T00:00:00.000Z') {
    return -Infinity;
  }
```

While your simplified version of the Infinity check would work in our scenario, since we mainly have either 1) Infinity timestamps or 2) e.g. "2022-02-13 00:00:00", but there is a flaw there in terms of supporting JS max timestamps.

In the current conditional, following date would yield Infinity value: "+275760-08-13 00:00:00" (is not actually Infinity) & the next would throw error since it is out of range: "-271821-05-20 00:00:00" (should be Infinity).

This is, because JS dates max date as ISO string is "+275760-09-13T00:00:00.000Z" and minimum "-271821-04-20T00:00:00.000Z" respectively (see the different months there). Therefore, I had added the year check first to ensure that the number of characters in ISO date string would be correct, thus, allowing to compare the ISO date strings only after we are sure, that we are not comparing years, but only the rest of the string.

> Date and time values are ordered from the largest to smallest unit of time: year, month (or week), day, hour, minute, second, and fraction of second. The [lexicographical order](https://en.wikipedia.org/wiki/Lexicographical_order) of the representation thus corresponds to chronological order, except for date representations involving negative years or time offset. This allows dates to be naturally [sorted](https://en.wikipedia.org/wiki/Sorting) by, for example, file systems.

This can be interpreted as ISO date strings are comparable if they are in the correct date range. Negative years won't raise an issue here, when the year check is made before comparing the dates.

Reply:
I will write unit tests for these scenarios and provide PR soon.